kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -304,12 +307,17 @@ private void setupContextMenuMenuItems() {
                 {"Save to Favorites",  false, new String[]{"menu-item"}, null, createGraphics("favorites-icon")},
                 {MenuHelper.SEPARATOR},
                 {"Add Comment",  false, new String[]{"menu-item"}, null, createGraphics("comment-icon")},
-                {"Remove", true, new String[]{"menu-item"}, null, createGraphics("remove-icon")}
+                {"Remove", true, new String[]{"menu-item"}, (EventHandler<ActionEvent>) this::removeField, createGraphics("remove-icon")}
         };
 
         contextMenu = MenuHelper.getInstance().createContextMenuWithMenuItems(menuItems);
     }
 
+    private void removeField(ActionEvent actionEvent) {
+        fieldsTilePane.getChildren().remove(selectedNodeField);
```

Comment:
Can we remove this line? This statement should be in the `FIELDS_COLLECTION` observable list change listener.

Reply:
**Can we wire things up as we discussed related to the two phases?**

1. Subscriber (listens for delete event),
2. ObservableList (view model's list) add change listener to update UI  
While the ui is the `fieldsTilePane.getChildren()` it may or may not have its own change listeners. In this case it does namely updateField numbers (order) on labels.

There should be an Event subscriber that listens for fields being deleted this would be responsible for removing the field from the  `patternViewModel.getObservableList(FIELDS_COLLECTION).remove(patternField)`.

a change listener on that observable list above listens for a remove, if so, removes it from the `fieldsTilePane.getChildren().remove(selectedNodeField);`